File: Largest_Substring_Between_Two_Equal_Characters.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        idx = [0] * 26
        res = -1

        for i in range(len(s)):
            if idx[ord(s[i]) - ord("a")] == 0:
                idx[ord(s[i]) - ord("a")] = i + 1
            res = max(res, i - idx[ord(s[i]) - ord("a")])
        
        return res

File: Largest_Substring_Between_Two_Equal_Characters.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        vector<int> idx(26, 0);
        int res = -1;

        for (int i = 0; i < s.size(); ++i) {
            if (idx[s[i] - 'a'] == 0) {
                idx[s[i] - 'a'] = i + 1;
            }
            res = max(res, i - idx[s[i] - 'a']);
        }

        return res;        
    }
};
